### ISSUE TITLE
Support for Click >= 8.0

### ISSUE THREAD

**Issue by jdbodyfelt** [aws/aws-sam-cli#3691]
### Description:

In a number of my travis builds, I'm getting errors relating to conflicts in versions of `click` used between `aws-sam-cli 1.39.0` and `black 22.1.0` (newest formatter) - the former requiring =~7.1 and the later with >=8.0.0. It would be great if that can be addressed on 1.39.1. 
A quick fix for me is rolling our black back to 20.8 or so, but we'd love to have most stable on that and aws-sam-cli. Thanks!

**Comment by mingkun2020** [comment]
Thanks for the feature request! We have a [roadmap](https://github.com/aws/aws-sam-cli/wiki/SAM-CLI-Roadmap) here, but we also try to take in popular feature requests based on 👍🏽 from the community. We will be adding this to our intake for prioritization.

**Comment by thyagram-aws** [comment]
Hi guys, we ran into issue with the 'click' python package as a dependency of aws-sam-cli (v1.45.0) while working on our solution: https://github.com/aws-solutions / [aws-virtual-waiting-room](https://github.com/aws-solutions/aws-virtual-waiting-room)

Could you look into upgrading the dependency version to >=8.0.0. Python safety package has version 7.1.2 flagged. 
https://pyupio.github.io/safety-db/ 

Running safety (v. 1.10.3) on a sample requirements.txt produces the following output. 
```
$ cat requirements.txt
click==7.1.2
$ safety check -r requirements.txt
+==============================================================================+
|                                                                              |
|                               /$$$$$$            /$$                         |
|                              /$$__  $$          | $$                         |
|           /$$$$$$$  /$$$$$$ | $$  \__//$$$$$$  /$$$$$$   /$$   /$$           |
|          /$$_____/ |____  $$| $$$$   /$$__  $$|_  $$_/  | $$  | $$           |
|         |  $$$$$$   /$$$$$$$| $$_/  | $$$$$$$$  | $$    | $$  | $$           |
|          \____  $$ /$$__  $$| $$    | $$_____/  | $$ /$$| $$  | $$           |
|          /$$$$$$$/|  $$$$$$$| $$    |  $$$$$$$  |  $$$$/|  $$$$$$$           |
|         |_______/  \_______/|__/     \_______/   \___/   \____  $$           |
|                                                          /$$  | $$           |
|                                                         |  $$$$$$/           |
|  by pyup.io                                              \______/            |
|                                                                              |
+==============================================================================+
| REPORT                                                                       |
| checked 1 packages, using free DB (updated once a month)                     |
+============================+===========+==========================+==========+
| package                    | installed | affected                 | ID       |
+============================+===========+==========================+==========+
| click                      | 7.1.2     | <8.0.0                   | 47833    |
+==============================================================================+
```

**Comment by mndeveci** [comment]
Thanks for raising this issue. But we highly recommend using our installers ([native linux/brew/msi](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)) for not experiencing these issues in the future.

If you still want to install it through `pip`, we would recommend setting up a virtual environment, specific to `aws-sam-cli` and install under there. In this way, you won't have any dependency conflicts in your project.
```
python3 -m venv .venv-sam-cli
source .venv-sam-cli/bin/activate
pip install aws-sam-cli
export PATH=$PATH:.venv-sam-cli/bin/sam # if you want to export executable globally
```

Please let us know if you have further questions.

**Comment by mndeveci** [comment]
Thanks for pointing out the security issue. We identified that it is only affecting usage of [pager feature](https://click.palletsprojects.com/en/8.1.x/utils/#pager-support) of click([link to commit](https://github.com/pallets/click/pull/1754/files)), and we confirm that AWS SAM CLI is [not using that feature in any of the commands as of now](https://github.com/aws/aws-sam-cli/search?q=echo_via_pager). So we can assure that this security issue is not affecting AWS SAM CLI users at the moment.

For the future incidents, please follow our guidelines to report security issues: https://github.com/aws/aws-sam-cli/blob/develop/CONTRIBUTING.md#security-issue-notifications

In the meantime, we are going to work on to upgrade the version of click library in the near future.

**Comment by sriram-mv** [comment]
https://github.com/aws/aws-sam-cli/pull/4326 has addressed this issue.

**Comment by mndeveci** [comment]
The fix for this issue is released with [1.61.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.61.0) version.
